### PR TITLE
Update Order definition and related logic & metrics

### PR DESCRIPTION
**Pull request by Jinyu-W**
# Description

- Add order expiration logic
- Add order statistics to env metrics (pending scheduled, expired order num, delayed scheduling, ...)
- Update order class definition, and enrich related logic
- Add order statistics to ConsumerUnit & DistributionUnit, and update processing logic
- Fix unit test issues

## Linked issue(s)/Pull request(s)

- [issue_number](issue_link)

## Type of Change

- [ ] Non-breaking bug fix
- [ ] Breaking bug fix
- [ ] New feature
- [ ] Test
- [ ] Doc update
- [ ] Docker update

## Related Component

- [ ] Simulation toolkit
- [ ] RL toolkit
- [ ] Distributed toolkit

## Has Been Tested

- OS:
  - [ ] Windows
  - [ ] Mac OS
  - [ ] Linux
- Python version:
  - [ ] 3.7
  - [ ] 3.8
  - [ ] 3.9
- Key information snapshot(s):

## Needs Follow Up Actions

- [ ] New release package
- [ ] New docker image

## Checklist

- [ ] Add/update the related comments
- [ ] Add/update the related tests
- [ ] Add/update the related documentations
- [ ] Update the dependent downstream modules usage

### CODE REVIEW

File: maro/simulator/scenarios/supply_chain/units/storage.py [hunk 1]
```diff
@@ -284,20 +284,23 @@ def try_add_products(
                     quantity = min(
                         int(quantity * fulfill_ratio_dict[storage_id]), self._remaining_space_dict[storage_id],
                     )
-                    self._add_product(sku_id, quantity)
-                    added_quantities[sku_id] = quantity
+                    if quantity > 0:
+                        self._add_product(sku_id, quantity)
+                        added_quantities[sku_id] = quantity
```
Comment (lihuoran) [inline maro/simulator/scenarios/supply_chain/units/storage.py:289]: How about move quantity check and `added_quantities[sku_id] = quantity` into `self._add_product`?

File: maro/simulator/scenarios/supply_chain/units/consumer.py [hunk 2]
```diff
@@ -51,46 +46,99 @@ def __init__(
...
             self.order_quantity_on_the_way[tick + i] for i in range(self.world.configs.settings["pending_order_len"])
         ]
-        if tick in self.order_quantity_on_the_way:  # Remove data at tick to save storage
-            self.order_quantity_on_the_way.pop(tick)
         return ret
 
-    @property
-    def in_transit_quantity(self) -> int:
-        return self._in_transit_quantity
+    def on_order_successfully_placed(self, order: Order) -> None:
```
Comment (lihuoran) [inline maro/simulator/scenarios/supply_chain/units/consumer.py:94]: I remember we had a discussion about this naming issue: a method called `on_XXX` should be bound with an event handler, and only be called when the handler is triggered. It should not be called actively. Similar to the following several methods.

File: maro/simulator/scenarios/supply_chain/units/storage.py (hunk 1 above)
Comment (Jinyu-W) [reply maro/simulator/scenarios/supply_chain/units/storage.py:289]: not fix here

File: maro/simulator/scenarios/supply_chain/units/consumer.py (hunk 2 above)
Comment (Jinyu-W) [reply maro/simulator/scenarios/supply_chain/units/consumer.py]: rename to `handle_xxx` instead

**Review by lihuoran — APPROVED** [review]

**Pull request merged**